Owlcat/FitsTool.py

```diff
@@ -66,18 +66,22 @@ def stack_planes(fitslist, outname='combined.fits', axis=0, ctype=None, keep_old
     crval = hdr['CRVAL%d' % fits_ind]
 
     imslice = [slice(None)] * naxis
-    _sorted = sorted([pyfits.open(fits) for fits in fitslist],
-                     key=lambda a: a[0].header['CRVAL%d' % (naxis - axis)])
+
+    # get headers, but don't keep the FITS files open (too many open files!)
+    _hdr_filenames = [(pyfits.open(fits)[0].header, fits) for fits in fitslist]
+
+    _sorted = sorted(_hdr_filenames,
+                     key=lambda a: a[0]['CRVAL%d' % (naxis - axis)])
 
     # define structure of new FITS file
-    nn = [hd[0].header['NAXIS%d' % (naxis - axis)] for hd in _sorted]
+    nn = [header['NAXIS%d' % (naxis - axis)] for header, _ in _sorted]
     shape = list(hdu.data.shape)
     shape[axis] = sum(nn)
     data = numpy.zeros(shape, dtype=float)
```

Comment:
Perhaps this should be  a memmap onto disk if the goal is to reduce memory footprint?